Neos.Flow/Tests/Unit/Configuration/ConfigurationManagerTest.php

```diff
@@ -784,6 +784,58 @@ public function replaceVariablesInPhpStringReplacesEnvMarkersWithEnvironmentValu
+        yield 'empty format skips replacement' => ['envVarName' => 'NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR', 'envVarValue' => '123', 'setting' => '%env():NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR%', 'expectedResult' => '%env():NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR%'];
+
+        yield 'format int casts non-numeric env variable to 0' => ['envVarName' => 'NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR', 'envVarValue' => 'foo', 'setting' => '%env(int):NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR%', 'expectedResult' => 0];
+        yield 'format int casts numeric env variable to its integer value' => ['envVarName' => 'NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR', 'envVarValue' => '123', 'setting' => '%env(int):NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR%', 'expectedResult' => 123];
+        yield 'integer-casted values can be concatenated with strings' => ['envVarName' => 'NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR', 'envVarValue' => '123', 'setting' => '%env(int):NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR%suffix', 'expectedResult' => '123suffix'];
+        yield 'integer-casted values can be concatenated with integers' => ['envVarName' => 'NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR', 'envVarValue' => '123', 'setting' => '%env(int):NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR%456', 'expectedResult' => '123456'];
+        yield 'format int casts non-existing env variable to 0' => ['envVarName' => '', 'envVarValue' => '', 'setting' => '%env(int):NEOS_FLOW_TESTS_UNIT_CONFIGURATION_NON_EXISTING_ENVIRONMENT_VARIABLE%', 'expectedResult' => 0];
+
+        yield 'format bool casts "1" to true' => ['envVarName' => 'NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR', 'envVarValue' => '1', 'setting' => '%env(bool):NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR%', 'expectedResult' => true];
+        yield 'format bool casts "true" to true' => ['envVarName' => 'NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR', 'envVarValue' => 'true', 'setting' => '%env(bool):NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR%', 'expectedResult' => true];
+        yield 'format bool casts "false" to true' => ['envVarName' => 'NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR', 'envVarValue' => 'false', 'setting' => '%env(bool):NEOS_FLOW_TESTS_UNIT_CONFIGURATION_CONFIGURATIONMANAGERTEST_MOCKENVVAR%', 'expectedResult' => true];
```

Comment:
wait a sec this a usecase to even write `"false"` in an env variable? If people do that i think we should cast it to false - otherwise this would be confusing

Reply:
I used PHPs default casting behavior.. But you're right. Maybe we should treat "false", "FALSE", "no", ... differently for boolean

Reply:
but as said if this is not good practice at all than we dont need it i think...

Reply:
> but as said if this is not good practice at all than we dont need it i think...

In fact that was the trigger for this issue: We used env Flags like "ENABLE_TRACKING=false" and then had to check for the actual string `false` in all places.
I think there is no harm in evaluating the string "false" to `false` as well if the type is set to `bool`